### PR TITLE
Master 20200414 003 request items

### PR DESCRIPTION
**Pull request by barranger**
So it works, but it's pretty ugly.  Going to add a card for styling, might not be for the MVP

### CODE REVIEW

File: collab-helper-web/src/screens/RequestItem.js [hunk 1]
```diff
@@ -16,11 +39,71 @@ const RequestItem = ({ tripId }) => {
...
+            </>
+          )}
+
+          {myTrip && (
+            <>
+              <Typography>
+                Here are the Items people have requested for you to pick up
+              </Typography>
+              <ul>
+                {trip.items.map((i) => (
```
Comment (annieEllenberger) [inline collab-helper-web/src/screens/RequestItem.js:86]: Running into a `TypeError: Cannot read property 'map' of undefined` here when navigating to a trip when:
1. It's the current user's trip
2. No one has requested items for the trip yet

File: collab-helper-web/src/services/tripService.js [hunk 2]
```diff
@@ -19,4 +20,11 @@ export const getTripById = async (id) => {
   const contactRef = firestore.doc(`trips/${id}`);
   const snapshot = await contactRef.get();
   return snapshot.data();
+}
+
+export const addItemToTrip = async (user, tripId, what) => {
+  if(!user) return;
+
+  const docRef = firestore.doc(`trips/${tripId}`);
+  await docRef.update({ items: firebase.firestore.FieldValue.arrayUnion({ user, what})})
```
Comment (annieEllenberger) [inline collab-helper-web/src/services/tripService.js:29]: Change `what` to something else... maybe `requestedItem`?

File: collab-helper-web/src/screens/RequestItem.js [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
 const RequestItem = ({ tripId }) => {
   const [loaded, setLoaded] = useState(false);
-  const [trip, setTrip] = useState();
+  const [trip, setTrip] = useState({});
+  const [item, setItem] = useState("");
+  const [open, setOpen] = useState(false);
+  const classes = useStyles();
+
+  const user = useContext(UserContext);
+
```
Comment (annieEllenberger) [inline collab-helper-web/src/screens/RequestItem.js:26]: Need to reload to show user their requested item (and prevent them from entering more until we have multi-request support in place)

File: collab-helper-web/src/services/tripService.js (hunk 2 above)
Comment (annieEllenberger) [inline collab-helper-web/src/services/tripService.js:29]: Can this be changed to *append* additional items requested in an array field associated with the same user?

File: collab-helper-web/src/screens/RequestItem.js (hunk 1 above)
Comment (annieEllenberger) [reply collab-helper-web/src/screens/RequestItem.js:86]: Also seeing the following React warning about unique "key" props for each item in the list here ...

File: collab-helper-web/src/screens/RequestItem.js (hunk 3 above)
Comment (barranger) [reply collab-helper-web/src/screens/RequestItem.js:55]: fixed in latest commit

File: collab-helper-web/src/services/tripService.js (hunk 2 above)
Comment (barranger) [reply collab-helper-web/src/services/tripService.js]: We will fix this as part of the next release with multi item requests

**Pull request merged**